include/openPMD/auxiliary/Filesystem.hpp

```diff
@@ -0,0 +1,45 @@
+/* Copyright 2018 Fabian Koller
+ *
+ * This file is part of openPMD-api.
+ *
+ * openPMD-api is free software: you can redistribute it and/or modify
+ * it under the terms of of either the GNU General Public License or
+ * the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * openPMD-api is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License and the GNU Lesser General Public License
+ * for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * and the GNU Lesser General Public License along with openPMD-api.
+ * If not, see <http://www.gnu.org/licenses/>.
+ */
+#pragma once
+
+#include <string>
+#include <vector>
+
+namespace openPMD
+{
+namespace auxiliary
+{
+bool
+path_exists( std::string const& path );
+
+bool
+file_exists( std::string const& path );
+
+std::vector< std::string >
+list_directory(std::string const& path );
```

Comment:
docstring: non-recursive, files-only?

Reply:
Non-recursive, every entry (i.e. directories and files).

Reply:
Makes sense. Best to just put this info in the `doxygen` doc string I guess.